### DISCUSSION TITLE
Input text `Mask` for example phone number

### DISCUSSION THREAD

**Discussion by ilomon10** [mantinedev/mantine#1544] [Feature requests]
Maybe can add `text masking` just like phone number, whenever we type `12345` it can automatic parse (mask) to `+44 123-45`, then whenever we type other char it can be ignored.

![chrome-capture-2022-4-30](https://user-images.githubusercontent.com/25215779/170978932-6c89f415-3841-4a55-b357-a77e93997581.gif)

we can easily implement any text formating with `+44 999-9999` or regex.
it can apply to `currency`, `date`, `decimal number`, `min-max number`, `email`, and any weird text format

![chrome-capture-2022-4-30 (1)](https://user-images.githubusercontent.com/25215779/170980365-e4467d3d-e40c-466f-83fc-f58b1e02a0ca.gif)

external module related:
- [imaskjs](https://github.com/uNmAnNeR/imaskjs)
- [react-input-mask](https://github.com/sanniassin/react-input-mask)

there was module that already done, but maybe we can embed direct with official support from mantine.

```jsx
// Example
<InputText format="+7 (000)000 0000" />
<InputText format="dd.mm yyyy" />
```

**Comment by rtivital** [comment]
You can use https://www.npmjs.com/package/react-input-mask with Input component. We will not add this feature as part of Mantine.

**Reply by ilomon10** [reply]
Ok thanks, thats quick response

**Reply by Toshinaki** [reply]
@rtivital 
Can we have some example showing how to add mask to Mantine Input?
Also I've found [this package](https://github.com/uNmAnNeR/imaskjs/tree/master/packages/react-imask) which is still maintained.

**Reply by rtivital** [reply]
We provide an example in documentation – https://mantine.dev/core/input/#custom-component

**Comment by goodlythink** [comment]
how to pass style Input to InputMask (react-input-mask)

**Reply by rtivital** [reply]
https://mantine.dev/core/input/#custom-component
